packages/devtools_app/lib/src/shared/object_tree.dart

```diff
@@ -0,0 +1,655 @@
+      // TODO: https://github.com/flutter/devtools/issues/2159
+      if (kIsWeb) {
+        return <DartObjectNode>[];
+      }
+      result = Int64List.view(bytes.buffer);
+      break;
+    case InstanceKind.kFloat32List:
+      result = Float32List.view(bytes.buffer);
+      break;
+    case InstanceKind.kFloat64List:
+      result = Float64List.view(bytes.buffer);
+      break;
+    case InstanceKind.kInt32x4List:
+      result = Int32x4List.view(bytes.buffer);
+      break;
+    case InstanceKind.kFloat32x4List:
+      result = Float32x4List.view(bytes.buffer);
+      break;
+    case InstanceKind.kFloat64x2List:
+      result = Float64x2List.view(bytes.buffer);
+      break;
+    default:
+      result = bytes;
+  }
+
+  for (int i = 0; i < result.length; i++) {
+    final name = instance.offset == null ? i : i + instance.offset!;
+    variables.add(
+      DartObjectNode.fromValue(
+        name: '[$name]',
+        value: result[i],
+        isolateRef: isolateRef,
+        artificialName: true,
+      ),
+    );
+  }
+  return variables;
+}
+
+List<DartObjectNode> _createVariablesForElements(
+  Instance instance,
+  IsolateRef? isolateRef,
+) {
+  final variables = <DartObjectNode>[];
+  final elements = instance.elements ?? [];
+  for (int i = 0; i < elements.length; i++) {
+    final name = instance.offset == null ? i : i + instance.offset!;
+    variables.add(
+      DartObjectNode.fromValue(
+        name: '[$name]',
+        value: elements[i],
+        isolateRef: isolateRef,
+        artificialName: true,
+      ),
+    );
+  }
+  return variables;
+}
+
+List<DartObjectNode> _createVariablesForFields(
+  Instance instance,
+  IsolateRef? isolateRef, {
+  Set<String>? existingNames,
+}) {
+  final variables = <DartObjectNode>[];
+  for (var field in instance.fields!) {
+    final name = field.decl!.name;
+    if (existingNames != null && existingNames.contains(name)) continue;
+    variables.add(
+      DartObjectNode.fromValue(
+        name: name,
+        value: field.value,
+        isolateRef: isolateRef,
+      ),
+    );
+  }
+  return variables;
+}
+
+// TODO(jacobr): gracefully handle cases where the isolate has closed and
+// InstanceRef objects have become sentinels.
+class DartObjectNode extends TreeNode<DartObjectNode> {
+  DartObjectNode._({
+    this.name,
+    this.text,
+    GenericInstanceRef? ref,
+    int? offset,
+    int? childCount,
+    this.artificialName = false,
+  })  : _ref = ref,
+        _offset = offset,
+        _childCount = childCount {
+    indentChildren = ref?.diagnostic?.style != DiagnosticsTreeStyle.flat;
+  }
+
+  /// Creates a variable from a value that must be an InstanceRef or a primitive
+  /// type.
+  ///
+  /// [value] should typically be an [InstanceRef] but can also be a [Sentinel]
+  /// [ObjRef] or primitive type such as num or String.
+  factory DartObjectNode.fromValue({
+    String? name,
+    required Object? value,
+    bool artificialName = false,
```

Comment:
Adding a description for `artificialName` might be helpful in the future adventures in this aread